packages/teleport-plugin-module-generator-angular/src/utils.ts

```diff
@@ -0,0 +1,172 @@
+  )
+
+  const exportsProperty: types.ObjectProperty = t.objectProperty(
+    t.identifier('exports'),
+    t.arrayExpression(componentsList)
+  )
+
+  return t.decorator(
+    t.callExpression(t.identifier('NgModule'), [
+      t.objectExpression([declerations, imports, exportsProperty]),
+    ])
+  )
+}
+
+export const createRootModuleDecorator = (t = types) => {
+  const declerations: types.ObjectProperty = t.objectProperty(
+    t.identifier('declarations'),
+    t.arrayExpression([t.identifier('AppComponent')])
+  )
+
+  const imports: types.ObjectProperty = t.objectProperty(
+    t.identifier('imports'),
+    t.arrayExpression([
+      t.identifier('BrowserModule'),
+      t.callExpression(t.memberExpression(t.identifier('RouterModule'), t.identifier('forRoot')), [
+        t.identifier('routes'),
+      ]),
+      t.identifier('ComponentsModule'),
+    ])
+  )
+
+  const providers = t.objectProperty(t.identifier('providers'), t.arrayExpression([]))
+
+  const bootstrap = t.objectProperty(
+    t.identifier('bootstrap'),
+    t.arrayExpression([t.identifier('AppComponent')])
+  )
+
+  return t.decorator(
+    t.callExpression(t.identifier('NgModule'), [
+      t.objectExpression([declerations, imports, providers, bootstrap]),
+    ])
+  )
+}
+
+export const createExportModuleAST = (moduleName: string, t = types) => {
+  return t.exportNamedDeclaration(
+    t.classDeclaration(t.identifier(moduleName), null, t.classBody([])),
+    [],
+    null
+  )
+}
+
+export const createRoutesAST = (routes, stateDefinitions, t = types) => {
+  // TODO: Need to generate type annotation for routes variable, currently babel throwing erro
+  const routesObject = routes.map((conditionalNode) => {
+    const { value: routeKey } = conditionalNode.content
+    const { fileName: pageName, path } = extractPageMetadata(
+      stateDefinitions.route,
+      routeKey.toString()
+    )
+
+    return t.objectExpression([
+      t.objectProperty(t.identifier('path'), t.stringLiteral(path.slice(1, path.length))),
+      t.objectProperty(
+        t.identifier('loadChildren'),
+        t.stringLiteral(
+          `./pages/${pageName}/${pageName}.module#${dashCaseToUpperCamelCase(`${pageName}-module`)}`
+        )
+      ),
+    ])
+  })
+
+  return t.variableDeclaration('const', [
+    t.variableDeclarator(t.identifier('routes'), t.arrayExpression(routesObject)),
+  ])
+}
+
+export const createPageRouteAST = (componentName: string, t = types) => {
+  return t.variableDeclaration('const', [
+    t.variableDeclarator(
+      t.identifier('routes'),
+      t.arrayExpression([
+        t.objectExpression([
+          t.objectProperty(t.identifier('path'), t.stringLiteral('')),
+          t.objectProperty(t.identifier('component'), t.identifier(componentName)),
+        ]),
+      ])
+    ),
+  ])
+}
+
+export const constructLocalDependency = (componentName: string) => {
+  const dependency: UIDLDependency = {
+    type: 'local',
+    path: `./${componentName}.component`,
```

Comment:
are these dependencies only inside the component folder? so they are not affected by the UIDL folder specification and the project strategy?

Reply:
No, as by default we push all the components into the components folder according to the folder structure.